### PR TITLE
Add more linting rules and apply lint fixes

### PR DESCRIPTION
**Pull request by badrange**

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -4,7 +4,33 @@
...
-        "@typescript-eslint/explicit-function-return-type": "off"
+        "@typescript-eslint/brace-style": ["error"],
+        "@typescript-eslint/explicit-function-return-type": "off",
+        "@typescript-eslint/func-call-spacing": ["error"],
+        "@typescript-eslint/member-ordering": ["warn", { "default": ["method", "constructor", "field"] }],
+        "@typescript-eslint/no-require-imports": ["error"],
+        "react/no-unused-prop-types": ["warn", { "skipShapeProps": true }],
+        "arrow-spacing": "warn",
+        "array-bracket-spacing": ["warn", "never"],
+        "brace-style": "off",
```
Comment (midori0507) [inline .eslintrc.json]: Why there is typescript rule `brace-style` is `error` above but global `brace-style` is `off` here

File: .eslintrc.json [hunk 2]
```diff
@@ -4,7 +4,33 @@
...
+          {
+            "groups": [
+              "builtin",
+              "external",
+              ["internal", "parent", "sibling", "index"]
+            ],
+            "newlines-between": "always"
+          }
+        ],
+        "max-len": [ "warn", { "code":  120}],
```
Comment (midori0507) [inline .eslintrc.json]: trim empty space before 120

Comment (midori0507) [reply .eslintrc.json]: no space opening array as well

File: .eslintrc.json [hunk 3]
```diff
@@ -1,11 +1,38 @@
...
+        "object-curly-spacing": ["warn", "always"],
+        "sort-keys": ["warn", "asc", { "caseSensitive": true, "natural": true, "minKeys": 2 }]
       }
     }
-  ]
+  ],
+  "plugins": ["prettier"],
+  "rules": {
+    "prettier/prettier": "error"
+  }
```
Comment (Nizar-Rahme) [inline .eslintrc.json]: This can be merged with the `rules` field above

Comment (midori0507) [reply .eslintrc.json]: We remove this by using recommended settings

**Pull request merged**